src/components/EmojiWindow/EmojiWindow.tsx

```diff
@@ -0,0 +1,171 @@
+
+import {
+  emojiRowContainerStyles,
+  emojiListContainerStyles,
+  searchInputContainerStyles,
+  tabTextStyles
+} from "./EmojiWindow.Styles";
+import Emoji, { EmojiDetails } from "../Emoji/Emoji";
+import emojiData from "../../emojis.json";
+import { EMOJIS_PER_ROW, EMOJI_HEIGHT, PICKER_WIDTH, PICKER_HEIGHT } from "../../constants";
+import SmileyIcon from "../icons/SmileyIcon";
+import CatIcon from "../icons/CatIcon";
+import BurgerIcon from "../icons/BurgerIcon";
+import BallIcon from "../icons/BallIcon";
+import TravelIcon from "../icons/TravelIcon";
+import LightbulbIcon from "../icons/LightbulbIcon";
+import PeaceIcon from "../icons/PeaceIcon";
+import FlagIcon from "../icons/FlagIcon";
+
+interface EmojiWindowProps {
+    onEmojiClick: (nativeEmoji: string) => void;
+}
+
+enum EmojiCategory {
+    all = "All",
+    smileysPeople = "Smileys & People",
+    animalsNature = "Animals & Nature",
+    foodDrink = "Food & Drink",
+    activity = "Activity",
+    travelPlaces = "Travel & Places",
+    objects = "Objects",
+    symbols = "Symbols",
+    flags = "Flags"
+}
+
+const categoryIconProps = {
+    decorative: true,
+    size: 10,
+    color: "transparent"
+};
+
+const EmojiWindow = (props: EmojiWindowProps) => {
+    const [filteredEmojis, setFilteredEmojis] = useState(emojiData);
+    const [rowCount, setRowCount] = useState(Math.ceil(emojiData.length / EMOJIS_PER_ROW));
+    const [selectedCategory, setSelectedCategory] = useState(EmojiCategory.all);
+    const [searchInput, setSearchInput] = useState("");
+
+    useEffect(() => {
+        setRowCount(Math.ceil(filteredEmojis.length / EMOJIS_PER_ROW));
+    }, [filteredEmojis]);
+
+    // Re-compute emoji list after search input or category tab changes
+    useEffect(() => {
+        setFilteredEmojis(
+            emojiData.filter(emoji =>
+                (selectedCategory === EmojiCategory.all || emoji.category === selectedCategory) &&
+                emoji.description.toLowerCase().includes(searchInput)
+            )
+        );
+    }, [searchInput, selectedCategory]);
+
+    const rowRenderer = ({ key, index, style }: { key: string; index: number; style: CSSProperties }) => {
+        const items = [];
+        let startIndex = index * EMOJIS_PER_ROW;
+        const endIndex = Math.min((index + 1) * EMOJIS_PER_ROW, filteredEmojis.length);
+
+        for (let currentIndex = startIndex; currentIndex < endIndex; currentIndex++) {    
+            const emoji = filteredEmojis[currentIndex] as EmojiDetails;
+            items.push(
+                <Emoji emoji={emoji} onClick={props.onEmojiClick} />
+            );
+        }
+
+        return <Box {...emojiRowContainerStyles} key={key} style={style}>{items}</Box>;
+    };
+
+    return (
+        <Box {...emojiListContainerStyles}>
+            <Tabs selectedId={selectedCategory} variant="fitted">
+                <TabList aria-label="State hook tabs">
+                    <Tab
```

Comment:
Could we put the category and title info in an array and map over it to create the tabs?

Reply:
I thought of doing something like this too, but its not just the category enum type, but also a different icon component too. So the array would be `[{ category: EmojiCategory.animalsAndNature, icon: CatIcon }, ...]`. Do you still think thats clearer? I'm on the fence.

Reply:
Hmm yeah on balance probably not worth changing then